presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -290,11 +291,13 @@ else if (conjunct instanceof ComparisonExpression) {
                     if (firstDependencies.stream().allMatch(left::canResolve) && secondDependencies.stream().allMatch(right::canResolve)) {
                         leftComparisonExpressions.add(firstExpression);
                         rightComparisonExpressions.add(secondExpression);
+                        addNullFilters(complexJoinExpressions, node.getType(), firstExpression, secondExpression);
```

Comment:
Why is the null filter added to `complexJoinExpressions` rather than `leftComparisonExpressions` or `rightComparisonExpressions`?

Reply:
@rongrong `leftComparisonExpressions` or `rightComparisonExpressions` have different meaning in this function. As of today, in current master branch (where  my changes are not added), if I run the query
```
explain SELECT * FROM customer a INNER JOIN orders b ON a.custkey = b.custkey and a.phone is not null;
```
`leftComparisonExpressions` is `a.custkey`, `rightComparisonExpressions` is `b.custkey` and `complexJoinExpressions` is `a.phone is not null`.